crates/rattler_installs_packages/src/venv.rs

```diff
@@ -0,0 +1,79 @@
+//! Module that helps with allowing in the creation of python virtual environments.
+//! Now just use the python venv command to create the virtual environment.
+//! Later on we can look into actually creating the environment by linking to the python library,
+//! and creating the necessary files. See: [VEnv](https://packaging.python.org/en/latest/specifications/virtual-environments/#declaring-installation-environments-as-python-virtual-environments)
+#![allow(dead_code)]
+use crate::utils::{python_executable, FindPythonError};
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum VEnvError {
+    #[error(transparent)]
+    FindPythonError(#[from] FindPythonError),
+    #[error("failed to run 'python -m venv': `{0}`")]
+    FailedToRun(String),
+    #[error(transparent)]
+    FailedToCreate(#[from] std::io::Error),
+}
+
+/// Represents a virtual environment
+/// that we can install wheels into
+pub struct VEnv {
+    location: PathBuf,
+}
+
+impl VEnv {
+    fn new(location: PathBuf) -> Self {
+        Self { location }
+    }
+}
+
+/// Specifies where to find the python executable
+pub enum PythonLocation {
+    /// Use system interpreter
+    System,
+    // Use custom interpreter
+    Custom(PathBuf),
+}
+
+impl PythonLocation {
+    /// Location of python executable
+    pub fn executable(&self) -> Result<PathBuf, FindPythonError> {
+        match self {
+            PythonLocation::System => python_executable(),
+            PythonLocation::Custom(path) => Ok(path.clone()),
+        }
+    }
+}
+
+/// Create a virtual environment at specified directory
+pub fn create_venv(venv_dir: &Path, python: PythonLocation) -> Result<VEnv, VEnvError> {
+    let python = python.executable()?;
+    let mut cmd = Command::new(python);
+    cmd.arg("-m");
+    cmd.arg("venv");
```

Comment:
FWIW, if you want to avoid the overhead of a subprocess call here, it's not particularly difficult to implement the logic for creating a virtual environment, especially given that you don't want to have any packages installed within it: https://peps.python.org/pep-0405/#creating-virtual-environments has the details.

Reply:
Would be nice since you no longer need a python executable then!

Reply:
Yeah good idea! I had a todo at the top of the module that said exactly that would like to do it in another PR. Want to get to the actual sdist building first. 😀 

@baszalmstra you still need the executable to link it into the environment. Or pull one of conda-forge in any case the venv needs one.